tests/ClosureTest.php

```diff
@@ -12,6 +13,36 @@ class ClosureTest extends TestCase
     use InvalidCodeAnalysisTestTrait;
     use ValidCodeAnalysisTestTrait;
 
+    public function testLanguageServerFirstClassCallable(): void
+    {
+        $code = '<?php
+            function foo(): void {}
+
+            class A {
+                public static function publicStatic(): void {}
+
+                public function publicMethod(): void {}
+            }
+
+            foo(...);
+            A::publicStatic(...);
+            (new A())->publicMethod(...);
+        ';
+
+        $context = new Context();
+
+        $this->project_analyzer->setPhpVersion('8.1.0', 'tests');
+
+        $codebase = $this->project_analyzer->getCodebase();
+        $codebase->enterServerMode();
```

Comment:
I couldn't see a way to add this to `providerValidCodeParse` because this `enterServerMode` call was necessary to reproduce the issue.

Reply:
Looking at how `Codebase::$store_node_types` is used, I think we should rather enable it for all tests.